bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/discovery/MiIoDiscovery.java

```diff
@@ -152,17 +224,22 @@ private void discovered(String ip, byte[] response) {
                 isOnline = cloudInfo.getIsOnline();
             }
         }
+        submitDiscovery(ip, token, id, label, country, isOnline);
+    }
+
+    private void submitDiscovery(String ip, String token, String id, String label, String country, boolean isOnline) {
         ThingUID uid = new ThingUID(THING_TYPE_MIIO, id);
-        logger.debug("Discovered Mi Device {} ({}) at {} as {}", id, Long.parseUnsignedLong(id, 16), ip, uid);
         DiscoveryResultBuilder dr = DiscoveryResultBuilder.create(uid).withProperty(PROPERTY_HOST_IP, ip)
                 .withProperty(PROPERTY_DID, id);
         if (IGNORED_TOKENS.contains(token)) {
+            logger.debug("Discovered Mi Device {} ({}) at {} as {}", id, Long.parseUnsignedLong(id, 16), ip, uid);
```

Comment:
Is it guaranteed that `id` is an integer at this point? You make no attempt to catch the NumberFormatException that this would throw. If you know that it is an integer why not have the `id` passed as a long parameter instead?

Reply:
it is a HEX string in this setting..
But I can indeed surround it with a try/catch NumberFormatException

Reply:
![image](https://user-images.githubusercontent.com/5871213/101224262-63a5d380-368e-11eb-936f-0cd6d5bcaf98.png)